### PR TITLE
Change property condition check to a function, add nestng support.

### PR DESCRIPTION
**Pull request by h2zero**
## Description:
Moves the property condition check to a new function. This allows for the use of nested arrays as the function can be called recursively.

Example (json array nesting):
```
"condition":[["manufacturerdata", 10, "8", "|", "manufacturerdata", 10, "9", "|", "manufacturerdata", 10, "a", "|", "manufacturerdata", 10, "b"], "&", "manufacturerdata", 14, "!", "ffff"]
```

* Added some macros for "servicedata" and "manufacturerdata" to make the code cleaner.

TODO: docs

## Checklist:
  - [x] The pull request is done against the latest development branch
  - [x] Only one feature/fix was added per PR and the code change compiles without warnings
  - [x] I accept the [DCO](https://github.com/theengs/decoder/blob/development/docs/participate/development.md#developer-certificate-of-origin).

### CODE REVIEW

**Comment by DigiH** [comment]
LGTM with passing all the extra NOT tests

**Comment by 1technophile** [comment]
Thanks guys!

File: src/devices/IBT_2X_json.h [hunk 1]
```diff
@@ -7,22 +7,22 @@ const char* _IBT_2X_json = "{\"brand\":\"Inkbird\",\"model\":\"iBBQ\",\"model_id
    "condition":["name", "index", 0, "iBBQ","&","manufacturerdata", ">=", 24, "index", 0, "0000","&","manufacturerdata", "<=", 28, "index", 0, "0000"],
    "properties":{
       "tempc":{
-         "condition":["manufacturerdata", 0, "00000000", "&", "manufacturerdata", 16, "!", "f6ff"],
```
Comment (1technophile) [inline src/devices/IBT_2X_json.h:10]: is it voluntary to impact IBT_2X conditions? I'm not understanding why
@h2zero

Comment (DigiH) [reply src/devices/IBT_2X_json.h:10]: Actually there still is an issue with the IBT_2X, and also BT_4XS decoder, with them needing to have their compressed const char* properly done. It was thanks to the IBT_2X that the multi byte-index issue became apparent. With fixing those two decoders I will reintroduce the NOT for their high/probe not connected temperatures.

Comment (h2zero) [reply src/devices/IBT_2X_json.h:10]: I can remove these changes, I used them from @DigiH's fork to test what he was testing. I forgot about them when I made the PR and should be separated to a new PR.

Comment (1technophile) [reply src/devices/IBT_2X_json.h:10]: okay got it

Comment (DigiH) [reply src/devices/IBT_2X_json.h:10]: The fixes which I preprepared for the IBT_2X are at

https://github.com/DigiH/decoder/blob/not-fixes/src/devices/IBT_2X_json.h

which now should finally have 

`"condition":["manufacturerdata", 0, "!", "00000000", "&", "manufacturerdata", 16, "!", "f6ff"],`

for tempc and tempc2. Which I think is the first double NOT AND condition

Comment (h2zero) [reply src/devices/IBT_2X_json.h:10]: Updated with @DigiH's fixes as a separate commit, squashed all the others.

Comment (1technophile) [reply src/devices/IBT_2X_json.h:10]: >`  "&", "manufacturerdata", 16, "!", "f6ff"],`

I think this check and the others like this one are missing on the branch isn'it?

Comment (DigiH) [reply src/devices/IBT_2X_json.h:10]: These were the changes I wanted to submit once this was merged, with additional tests for the previous problematic IBT-2X - cleaned up now in one commit. Maybe @h2zero can grab this and included it here with this PR.

https://github.com/DigiH/decoder/commit/0fb6a9392e8164ff42c513fcc7ed2ffd6e1e0bbe

Comment (1technophile) [reply src/devices/IBT_2X_json.h:10]: @h2zero could you either remove the IBT_2X changes or pick the changes from here please https://github.com/DigiH/decoder/commit/0fb6a9392e8164ff42c513fcc7ed2ffd6e1e0bbe

Comment (DigiH) [reply src/devices/IBT_2X_json.h:10]: Probably easiest if I cherry pick this over, if that's ok.

Comment (1technophile) [reply src/devices/IBT_2X_json.h:10]: Yes I think so

Comment (DigiH) [reply src/devices/IBT_2X_json.h:10]: Done

**Comment by 1technophile** [comment]
Thanks!

File: src/devices/IBT_2X_json.h (hunk 1 above)
Comment (h2zero) [reply src/devices/IBT_2X_json.h:10]: Cool, thanks! I wasn't sure what specific changes you were referring to, I only used the ones in the link provided.

**Pull request merged**